### PR TITLE
Add dictionary decompounder docs #7979

### PR DESCRIPTION
**Pull request by AntonEliatra**
### Description
Add dictionary decompounder token filter docs

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/7979

### Version
All

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kolchfa-aws** [comment]
@udabhas Could you review this PR for technical accuracy?

**Comment by vagimeli** [comment]
@varun-lodaya The documentation is awaiting tech review and approval, which is delaying progress. Could you please suggest alternative reviewers who can assist with this task in a timely manner? We're eager to move this forward. Thank you.

**Comment by vagimeli** [comment]
> @varun-lodaya The documentation is awaiting tech review and approval, which is delaying progress. Could you please suggest alternative reviewers who can assist with this task in a timely manner? We're eager to move this forward. Thank you.

@varun-lodaya This is over a month old. We need tech review approval to move it forward in the documentation process. Please review this week or provide a peer who can review it. Thank you.

**Review by kolchfa-aws — COMMENTED** [review]
Thank you, @AntonEliatra! Please see my comments and let me know if you have any questions.

**Review by kolchfa-aws — APPROVED** [review]
LGTM

File: _analyzers/token-filters/dictionary-decompounder.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
+---
+layout: default
+title: Dictionary decompounder
+parent: Token filters
+nav_order: 110
+---
+
+# Dictionary decompounder token filter
+
+The `dictionary_decompounder` token filter is used to split compound words into their constituent parts based on a predefined dictionary. This filter is particularly useful in languages like German, Dutch, or Finnish, where compound words are common and breaking them down can improve search relevance. The `dictionary_decompounder` token filter works by taking each token (word) and checking if it can be split into smaller tokens based on a list of known words. If the token can be split into known words, the filter generates the subtokens for the token.
```
Comment (natebower) [inline _analyzers/token-filters/dictionary-decompounder.md:10]:
```suggestion
The `dictionary_decompounder` token filter is used to split compound words into their constituent parts based on a predefined dictionary. This filter is particularly useful for languages like German, Dutch, or Finnish, in which compound words are common, so breaking them down can improve search relevance. The `dictionary_decompounder` token filter determines whether each token (word) can be split into smaller tokens based on a list of known words. If the token can be split into known words, the filter generates the subtokens for the token.
```

File: _analyzers/token-filters/dictionary-decompounder.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+## Parameters
+
+The `dictionary_decompounder` token filter has the following parameters.
+
+Parameter | Required/Optional | Data type | Description
+:--- | :--- | :--- | :--- 
+`word_list` | Required unless `word_list_path` is configured | Array of strings | The dictionary containing words that the filter uses to split compound words into.
```
Comment (natebower) [inline _analyzers/token-filters/dictionary-decompounder.md:18]:
```suggestion
`word_list` | Required unless `word_list_path` is configured | Array of strings | The dictionary of words that the filter uses to split compound words.
```

File: _analyzers/token-filters/dictionary-decompounder.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+`word_list_path` | Required unless `word_list` is configured | String | A file path to a text file containing the dictionary words. Accepts either an absolute path or a path relative to the `config` directory. The dictionary file must be UTF-8 encoded and must contain each word on its own line.
```
Comment (natebower) [inline _analyzers/token-filters/dictionary-decompounder.md:19]:
```suggestion
`word_list_path` | Required unless `word_list` is configured | String | A file path to a text file containing the dictionary words. Accepts either an absolute path or a path relative to the `config` directory. The dictionary file must be UTF-8 encoded, and each word must be listed on a separate line.
```

File: _analyzers/token-filters/dictionary-decompounder.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+`min_word_size` | Optional | Integer | The minimum length of the entire compound word that will be considered for splitting. If a compound word is shorter than this value, it is not split. Default is `5`.
+`min_subword_size` | Optional | Integer | The minimum length for any subword. If a subword is shorter than this value, it is not included in the output. Default is `2`.
+`max_subword_size` | Optional | Integer | The maximum length for any subword. If a subword is longer than this value, it is not included in the output. Default is `15`.
+`only_longest_match` | Optional | Boolean | If set to `true`, only the longest matching subword will be returned. Default is `false`.
+
+## Example
+
+The following example request creates a new index named `decompound_example` and configures an analyzer with `dictionary_decompounder` filter:
```
Comment (natebower) [inline _analyzers/token-filters/dictionary-decompounder.md:27]:
```suggestion
The following example request creates a new index named `decompound_example` and configures an analyzer with the `dictionary_decompounder` filter:
```

File: _analyzers/token-filters/index.md [hunk 5]
```diff
@@ -23,7 +23,7 @@ Token filter | Underlying Lucene token filter|  Description
...
-`dictionary_decompounder` | [DictionaryCompoundWordTokenFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/compound/DictionaryCompoundWordTokenFilter.html) | Decomposes compound words found in many Germanic languages.
+[`dictionary_decompounder`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/dictionary-decompounder/) | [DictionaryCompoundWordTokenFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/compound/DictionaryCompoundWordTokenFilter.html) | Decomposes compound words found in many Germanic languages.
```
Comment (natebower) [inline _analyzers/token-filters/index.md:26]:
```suggestion
[`dictionary_decompounder`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/dictionary-decompounder/) | [DictionaryCompoundWordTokenFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/compound/DictionaryCompoundWordTokenFilter.html) | Splits compound words found in many Germanic languages.
```

File: _analyzers/token-filters/index.md [hunk 6]
```diff
@@ -23,7 +23,7 @@ Token filter | Underlying Lucene token filter|  Description
...
 `edge_ngram` | [EdgeNGramTokenFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/ngram/EdgeNGramTokenFilter.html) | Tokenizes the given token into edge n-grams (n-grams that start at the beginning of the token) of lengths between `min_gram` and `max_gram`. Optionally, keeps the original token.
```
Comment (natebower) [inline _analyzers/token-filters/index.md:27]: Line 26: Should this match the description in the preceding file? "Splits compound words into their constituent parts based on a predefined dictionary"? That they are found in many Germanic languages seems like supplementary information (and we position it that way in the preceding file).

**Review by natebower — COMMENTED** [review]
@kolchfa-aws @AntonEliatra Please see my comment and changes and let me know if you have any questions. Thanks!

**Pull request merged**